spacy_llm/tasks/srl_task.py

```diff
@@ -0,0 +1,407 @@
+
+        pred_relation_tuples.update(
+            [(i, ArgRELItem(**dict(r))) for r in pred_doc._.relations]
+        )
+        gold_relation_tuples.update(
+            [(i, ArgRELItem(**dict(r))) for r in gold_doc._.relations]
+        )
+
+    def _overlap_prf(gold: set, pred: set):
+        overlap = gold.intersection(pred)
+        p = 0.0 if not len(pred) else len(overlap) / len(pred)
+        r = 0.0 if not len(gold) else len(overlap) / len(gold)
+        f = 0.0 if not p or not r else 2 * p * r / (p + r)
+        return p, r, f
+
+    predicates_prf = _overlap_prf(gold_predicates_spans, pred_predicates_spans)
+    micro_rel_prf = _overlap_prf(gold_relation_tuples, pred_relation_tuples)
+
+    def _get_label2rels(rel_tuples: Iterable[Tuple[int, ArgRELItem]]):
+        label2rels = defaultdict(set)
+        for tup in rel_tuples:
+            label_ = tup[1].label
+            label2rels[label_].add(tup)
+        return label2rels
+
+    pred_label2relations = _get_label2rels(pred_relation_tuples)
+    gold_label2relations = _get_label2rels(gold_relation_tuples)
+
+    all_labels = set.union(
+        set(pred_label2relations.keys()), set(gold_label2relations.keys())
+    )
+    label2prf = {}
+    for label in all_labels:
+        pred_label_rels = pred_label2relations[label]
+        gold_label_rels = gold_label2relations[label]
+        label2prf[label] = _overlap_prf(gold_label_rels, pred_label_rels)
+
+    return {
+        "Predicates": predicates_prf,
+        "ARGs": {"Overall": micro_rel_prf, "PerLabel": label2prf},
+    }
+
+
+@registry.llm_tasks("spacy.SRL.v1")
+def make_srl_task(
+    labels: Union[List[str], str] = [],
+    template: str = _DEFAULT_SPAN_SRL_TEMPLATE_V1,
+    label_definitions: Optional[Dict[str, str]] = None,
+    examples: ExamplesConfigType = None,
+    normalizer: Optional[Callable[[str], str]] = None,
+    alignment_mode: Literal["strict", "contract", "expand"] = "contract",
+    case_sensitive_matching: bool = True,
+    single_match: bool = True,
+    verbose: bool = False,
+    predicate_key: str = "Predicate",
+):
+    """SRL.v1 task factory.
+
+    labels (str): Comma-separated list of labels to pass to the template.
+        Leave empty to populate it at initialization time (only if examples are provided).
+    template (str): Prompt template passed to the model.
+    label_definitions (Optional[Dict[str, str]]): Map of label -> description
+        of the label to help the language model output the entities wanted.
+        It is usually easier to provide these definitions rather than
+        full examples, although both can be provided.
+    examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+        reads a file containing task examples for few-shot learning. If None is
+        passed, then zero-shot learning will be used.
+    normalizer (Optional[Callable[[str], str]]): optional normalizer function.
+    alignment_mode (str): "strict", "contract" or "expand".
```

Comment:
```suggestion
    alignment_mode (Literal["strict", "contract", "expand"]): "strict", "contract" or "expand".
```
And the description should refer to what it does (I know this is not on you, we have that in other places already unfortunately).

Reply:
done. https://github.com/explosion/spacy-llm/pull/226/commits/be5065570bd235e5b297f85228770cd100c49ff8